src/Angular.js

```diff
@@ -840,43 +842,48 @@ function copy(source, destination) {
 
     stackSource.push(source);
     stackDest.push(destination);
-    return copyRecurse(source, destination);
+    return copyRecurse(source, destination, currentDepth);
   }
 
-  return copyElement(source);
+  return copyElement(source, currentDepth);
+
+  function copyRecurse(source, destination, currentDepth) {
+    currentDepth++;
+    if (maxDepth !== -1 && currentDepth > maxDepth) {
+      return typeof source;
```

Comment:
I would prefer some less ambiguous value that also makes it clear there was more stuff that has been truncated. Also, `typeof` can be misleading or at least not so useful, because almost anything that is not a primitive will return `object`.

If we wanted to keep it simple, maybe something like `'...'`. If we wanted to get more fancy, trying to get the name of the value's constructor and returning something like `constructorName + '(...)'` would be nice, but probably not worth it.